LEAF_Request_Portal/admin/templates/mod_templates_reports.tpl

```diff
@@ -3,265 +3,970 @@
+            data = codeEditor.edit.getValue();
+        } else if (typeof codeEditor.getValue === 'function') {
+            data = codeEditor.getValue();
+        }
 
-function save() {
-	$('#saveIndicator').attr('src', '../images/indicator.gif');
-	$.ajax({
-		type: 'POST',
-		data: {CSRFToken: '<!--{$CSRFToken}-->',
-			   file: codeEditor.getValue()},
-        url: '../api/reportTemplates/_' + currentFile,
-		success: function(res) {
-			$('#saveIndicator').attr('src', '../dynicons/?img=media-floppy.svg&w=32');
-			$('.modifiedTemplate').css('display', 'block');
-			var time = new Date().toLocaleTimeString();
-			$('#saveStatus').html('<br /> Last saved: ' + time);
-            if(res != null) {
-                alert(res);
-            }
-		}
-	});
-}
-
-function newReport() {
-    dialog.setTitle('New File');
-    dialog.setContent('Filename: <input type="text" id="newFilename"></input>');
+        if (data === currentFileContent) {
+            alert('There are no changes to save.');
+            return;
+        }
 
-    dialog.setSaveHandler(function() {
-    	var file = $('#newFilename').val();
         $.ajax({
             type: 'POST',
-            url: '../api/reportTemplates',
-            data: {CSRFToken: '<!--{$CSRFToken}-->',
-            	filename: file},
+            data: {
+                CSRFToken: '<!--{$CSRFToken}-->',
+                file: data
+            },
+            url: '../api/reportTemplates/_' + currentFile,
             success: function(res) {
-            	if(res == 'CreateOK') {
-                    updateFileList();
-            		loadContent(file);
-            	}
-            	else {
-            	    alert(res);
-            	}
+                var time = new Date().toLocaleTimeString();
+                $('#saveStatus').html('<br /> Last saved: ' + time);
+                setTimeout(function() {
+                    $('#saveStatus').fadeOut(1000, function() {
+                        $(this).html('').fadeIn();
+                    });
+                    loadContent(currentFile);
+                }, 3000);
+                saveFileHistory();
+            },
+            error: function() {
+                alert('An error occurred while saving the file.');
+            },
+            complete: function() {
+                $('#saveIndicator').attr('src', '');
             }
         });
-        dialog.hide();
+    }
+
+    function saveFileHistory() {
+        $.ajax({
+                type: 'POST',
+                data: {CSRFToken: '<!--{$CSRFToken}-->',
+                file: codeEditor.getValue()
+            },
+            url: '../api/reportTemplates/fileHistory/_' + currentFile,
+            success: function(res) {
+                console.log("File history has been saved.");
+                getFileHistory(currentFile);
+            }
+        });
+    }
+
+    function formatFileSize(bytes, threshold = 1024) {
+        if (bytes < threshold) {

```

Comment:
This could probably be done with a while loop but it's fine for now.

Reply:
Have made the change to a while loop